main.go

```diff
@@ -31,6 +32,20 @@ var javaExcludePrefixes = []string{
 
 var log = logging.MustGetLogger()
 
+// countFunc returns the number of elements in s that satisfy f.
+func countFunc[S []E, E comparable](s S, f func(e E) bool) int {
```

Comment:
this is cute but it feels maybe a bit overly generic for the case we have here...

we could probably just do
```
if len(slices.DeleteFunc([]string{
		opts.Zip.Preamble,
		opts.Zip.PreambleFrom,
		opts.Zip.PreambleFile,
	}, func(s string) bool { return s != "" })) > 1 {
```

Reply:
Much nicer!

Reply:
`s != ""` needs to be `s == ""` because elements are deleted when `del` is true, but this is much cleaner.